src/library_webgpu.js

```diff
@@ -1702,10 +1720,16 @@ var LibraryWebGPU = {
     bufferWrapper.onUnmap = [];
     var buffer = bufferWrapper.object;
 
+    // Handle the defaulting of size required by WebGPU
+    var bufferSize = (new Uint32Array([size]))[0];
```

Comment:
is this necessary? the C header defines `size` as a `size_t` which afaict should be equivalent to `uint32_t` (because these bindings currently assume 32-bit WASM). So a WGPU_WHOLE_MAP_SIZE in C should be giving you `size === 0xFFFFFFFF`. Are you getting `-1` or something instead?

Reply:
Yep. console.log gives -1 and if I do `if( size === 0xFFFFFFFF)` directly the branch is not hit

Reply:
I kinda remember now we had a similar problem before, looks like it was the SetIndexBuffer and SetVertexBuffer commands. They call makeU64ToNumberWithSentinelAsUndefined which does stuff with -1 as well. makeU64ToNumberWithSentinelAsUndefined is also used by wgpuDeviceCreateBindGroup but that usage looks wrong because the values come from makeGetU32, not function args. Seems very likely to be broken code. In fact, most likely a bunch of usages of `makeU64ToNumber` break down after `2**31`.

anyway... how about just check `size === -1` here for now and leave a comment that the size seems to come in as int32_t.

Reply:
Sounds good if that is acceptable. This way we can avoid some performance overhead